src/edge/edgeManager.ts

```diff
@@ -383,12 +383,13 @@ export class EdgeManager {
         const modules = templateJson.modulesContent.$edgeAgent["properties.desired"].modules;
         let moduleName: string = "";
         let moduleInfo: ModuleInfo;
+        const marketplace = Marketplace.getInstance(this.context, Object.keys(modules));

```

Comment:
move this line into MARKETPLACE_MODULE block?

Reply:
Have you tested the following scenario
1. Open two edge solution in the workspace.
2. Add module for the first solution with market place. But don't continue to import
3. Add module for the second solution with marketplace. And continue to import
4. Check whether the module added to the correct target?